integration_test/third_party_apps_test.go

```diff
@@ -172,65 +172,61 @@ func installAgent(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.
 	return nonRetryable, agents.InstallPackageFromGCS(ctx, logger, vm, agents.OpsAgentType, packagesInGCS)
 }
 
-// expectedLogs encodes a series of assertions about what data we expect
-// to see in the logging backend.
-type expectedLogs struct {
-	// Note on tags: the "yaml" tag specifies the name of this field in the
-	// .yaml file.
-	LogEntries []expectedLog `yaml:"log_entries"`
+type logFields struct {
+	Name        string `yaml:"name" validate:"required"`
+	Value       string `yaml:"value" validate:"required"`
+	Type        string `yaml:"type" validate:"required"`
+	Description string `yaml:"description" validate:"optional,omitempty"`
 }
+
 type expectedLog struct {
-	LogName string `yaml:"log_name"`
-	// Map of field name to a regex that is expected to match the field value.
-	// For example, {"jsonPayload.message": ".*access denied.*"}.
-	FieldMatchers map[string]string `yaml:"field_matchers"`
+	LogName string       `yaml:"log_name" validate:"required"`
+	Fields  []*logFields `yaml:"fields" validate:"required"`
+}
+
+type integrationMetadata struct {
+	ExpectedLogs    []*expectedLog          `yaml:"expected_logs"`
```

Comment:
Same, these don't need to be pointers

Reply:
Also, how does it behave if `expected_logs` or `expected_metrics` is missing? The nil check in `runSingleTest` suggests this should be allowed (i.e. they are optional) but there's no `validate:"optional"` here. Not sure if the marshaler will give an error or just give you an empty slice.

Reply:
For activemq which only has metrics but no logs at this time, it will not fall under thiis code snippet so it will be no testing for logs
```
		if metadata.ExpectedMetrics != nil {
			logger.ToMainLog().Println("found expectedMetrics, running metrics test cases...")
			if err = runMetricsTestCases(ctx, logger, vm, metadata.ExpectedMetrics); err != nil {
				return nonRetryable, err
			}
		}
```
There is actually no built-in "optional" tag from golang validator: https://pkg.go.dev/github.com/go-playground/validator/v10. So i will remove the optional tag i put in else where